roles/openshift_master/templates/master.yaml.v1.j2

```diff
@@ -96,6 +96,24 @@ kubernetesMasterConfig:
   - v1beta3
   - v1
 {% endif %}
+{% if openshift.master.builddefaults_http_proxy or openshift.master.builddefaults_https_proxy
+   and openshift.common.version_gte_3_2_or_1_2 %}
+  admissionConfig:
+    pluginConfig:
+      BuildDefaults:
+        configuration:
+          apiVersion: v1
+          kind: BuildDefaultsConfig
+          gitHTTPProxy: {{ openshift.master.builddefaults_git_http_proxy }}
+          gitHTTPSproxy: {{ openshift.master.builddefaults_git_https_proxy }}
+          env:
+          - name: HTTP_PROXY
+            value: '{{ openshift.master.builddefaults_http_proxy }}'
+          - name: HTTPS_PROXY
+            value: '{{ openshift.master.builddefaults_https_proxy }}'
+          - name: NO_PROXY
+            value: '{{ openshift.master.builddefaults_no_proxy }}'
```

Comment:
I'm probably going to need to revisit the defaults we stuff into the build environment. When testing a build today it dawned on me that we're going to be leaking a lot of infrastructure information into the build environment that's probably not necessary. That said, the web-ui already exposes the node's hostname so this can probably slide for now. I'll look at it some more.